GeneralsMD/Code/GameEngine/Source/Common/CommandLine.cpp

```diff
@@ -1361,61 +1264,150 @@ static CommandLineParam params[] =
 	//{ "-allAdvice", parseAllAdvice },
 
 #if defined(RTS_DEBUG) || defined(RTS_INTERNAL) || defined(_ALLOW_DEBUG_CHEATS_IN_RELEASE)
-  { "-preload", parsePreload },
+	{ "-preload", parsePreload },
 #endif
 
 
 };
 
-// parseCommandLine ===========================================================
-/** Parse command-line parameters. */
-//=============================================================================
-void parseCommandLine(int argc, char *argv[])
+char *nextParam(char *newSource, const char *seps)
 {
-	// To parse command-line parameters, we loop through a table holding arguments
-	// and functions to handle them.  Comparisons can be case-(in)sensitive, and
-	// can check the entire string (for testing the presence of a flag) or check
-	// just the start (for a key=val argument).  The handling function can also
-	// look at the next argument(s), to accomodate multi-arg parameters, e.g. "-p 1234".
-	int arg=1, param;
-	Bool found;
+	static char *source = NULL;
+	if (newSource)
+	{
+		source = newSource;
+	}
+	if (!source)
+	{
+		return NULL;
+	}
+
+	// find first separator
+	char *first = source;//strpbrk(source, seps);
+	if (first)
+	{
+		// go past separator
+		char *firstSep = strpbrk(first, seps);
+		char firstChar[2] = {0,0};
+		if (firstSep == first)
+		{
+			firstChar[0] = *first;
+			while (*first == firstChar[0]) first++;
+		}
+
+		// find end
+		char *end;
+		if (firstChar[0])
+			end = strpbrk(first, firstChar);
+		else
+			end = strpbrk(first, seps);
+
+		// trim string & save next start pos
+		if (end)
+		{
+			source = end+1;
+			*end = 0;
+
+			if (!*source)
+				source = NULL;
+		}
+		else
+		{
+			source = NULL;
+		}
+
+		if (first && !*first)
+			first = NULL;
+	}
+
+	return first;
+}
+
+static void parseCommandLine(const CommandLineParam* params, int numParams)
+{
+	std::vector<char*> argv;
+
+	std::string cmdLine = GetCommandLineA();
+	char *token = nextParam(&cmdLine[0], "\" ");
+	while (token != NULL)
+	{
+		argv.push_back(strtrim(token));
+		token = nextParam(NULL, "\" ");	   
+	}
+	int argc = argv.size();
+
+	int arg = 0;
 
 #ifdef DEBUG_LOGGING
 	DEBUG_LOG(("Command-line args:"));
 	int debugFlags = DebugGetFlags();
 	DebugSetFlags(debugFlags & ~DEBUG_FLAG_PREPEND_TIME); // turn off timestamps
-	for (arg=1; arg<argc; arg++)
+	for (arg=0; arg<argc; arg++)
```

Comment:
Is this change correct? Arg 0 refers to the executable path, not the first argument.

![image](https://github.com/user-attachments/assets/c6b5ab43-44e2-409a-994b-65a978491a65)